testing/test_basic_api.py

```diff
@@ -135,3 +135,24 @@ def parse(root):
 
     with pytest.raises(TypeError):
         setuptools_scm.get_version(parse=parse)
+
+
+def test_custom_version_cls():
+    """Test that `normalize` and `version_cls` work as expected"""
+
+    class MyVersion:
+        def __init__(self, tag_str: str):
+            self.version = tag_str
+
+        def __repr__(self):
+            return f"hello,{self.version}"
+
+    # you can not use normalize=False and version_cls at the same time
+    with pytest.raises(ValueError):
+        setuptools_scm.get_version(normalize=False, version_cls=MyVersion)
+
+    # TODO unfortunately with PRETEND_KEY the preformatted flag becomes True
+    #  which bypasses our class. which other mechanism would be ok to use here
+    #  to create a test?
+    # monkeypatch.setenv(setuptools_scm.PRETEND_KEY, "1.0.1")
+    # assert setuptools_scm.get_version(version_cls=MyVersion) == "1"
```

Comment:
I am not able to find a proper way to pass a version that I control, without entering in the "preformatted=True" bypass (and therefore not using the custom class). If you have an idea let me know. I would basically copy here the same kind of test that is done in git.

If you think that this is overkill, I can just remove this comment

Reply:
pretend-key is unable to pass versions, predent key is specifically intended to go for preformatted

Reply:
the basic way to do this is currently to use a number of inner apis (aka setting up config and meta, 

maybe izts time to ease testing a bit